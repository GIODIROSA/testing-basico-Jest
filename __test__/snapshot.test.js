import { getCharacter } from "../snaphot";
import rick from "../rick.json";

describe("Es hora de las instaneas", () => {
  test("Snapshot", () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test("Siempre fallara la instantanea", () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20),
    };
    expect(user).toMatchSnapshot();
  });

  test("Tenemos una excepsion del codigo", () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "Oscar Barajas",
    };
    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    });
  });
});

//para actualizar el snapshot... usar el comando jest -u
