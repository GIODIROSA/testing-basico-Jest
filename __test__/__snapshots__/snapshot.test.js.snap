// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instaneas Siempre fallara la instantanea 1`] = `
Object {
  "createAt": 2020-10-02T14:43:52.291Z,
  "id": 14,
}
`;

exports[`Es hora de las instaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instaneas Tenemos una excepsion del codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Oscar Barajas",
}
`;
